optuna/trial/_frozen.py

```diff
@@ -321,8 +322,15 @@ def _validate(self) -> None:
                     "`datetime_complete` is supposed to be None for an unfinished trial."
                 )
 
+        if self.state == TrialState.PRUNED and self._values is not None:
+            raise ValueError("`value` is supposed not to be set for a pruned trial.")
+        if self.state == TrialState.FAIL and self._values is not None:
+            raise ValueError("`value` is supposed not to be set for a failed trial.")
```

Comment:
```suggestion
        if self.state in (TrialState.PRUNED, TrialState.FAIL) and self._values is not None:
            raise ValueError(f"values should be None for a trial with state {self.state}, but got {self._values}.")
```
[nit] Make the code readable and easier to debug for users.